src/iterable/operators/skip.ts

```diff
@@ -16,13 +17,18 @@ export class SkipIterable<TSource> extends IterableX<TSource> {
     const it = this._source[Symbol.iterator]();
     let count = this._count;
     let next;
-    while (count > 0 && !(next = it.next()).done) {
-      count--;
-    }
-    if (count <= 0) {
-      while (!(next = it.next()).done) {
-        yield next.value;
+
+    try {
+      while (count > 0 && !(next = it.next()).done) {
+        count--;
+      }
+      if (count <= 0) {
+        while (!(next = it.next()).done) {
+          yield next.value;
+        }
       }
+    } finally {
+      returnIterator(it);
```

Comment:
This PR has me wondering if the style of explicitly constructing and enumerating the iterator should be avoided.

Perhaps we should refactor to be something like this?

```ts
let count = this._count;
for (const value of this._source) {
  if (--count < 0) {
    yield value;
  }
}
```

Reply:
I fully agree with that. Virtually every manual call to `[Symbol.(async)Iterator]()` is susceptible to this issue.

That's for another MR, though. I need this change to go through first (it's left a big "TODO: update ix" at work), but could help with fixing the remaining ones after this.